airbyte-webapp/src/hooks/services/ConfirmationModal/reducer.ts

```diff
@@ -0,0 +1,31 @@
+import { ActionType, createAction, createReducer } from "typesafe-actions";
+
+import { ConfirmationModalOptions, ConfirmationModalState } from "./types";
+
+export const actions = {
+  openConfirmationModal: createAction("OPEN_CONFIRMATION_MODAL")<ConfirmationModalOptions>(),
+  closeConfirmationModal: createAction("CLOSE_CONFIRMATION_MODAL")(),
+};
+
+type Actions = ActionType<typeof actions>;
+
+export const initialState: ConfirmationModalState = {
+  isOpen: false,
+  confirmationModal: null,
+};
+
+export const confirmationModalServiceReducer = createReducer<ConfirmationModalState, Actions>(initialState)
+  .handleAction(actions.openConfirmationModal, (state, action): ConfirmationModalState => {
+    return {
+      ...state,
+      isOpen: true,
+      confirmationModal: action.payload,
+    };
+  })
+  .handleAction(actions.closeConfirmationModal, (state): ConfirmationModalState => {
+    return {
+      ...state,
+      isOpen: false,
+      confirmationModal: null,
+    };
+  });
```

Comment:
Unless I'm misunderstanding something, it seems like this could be simplified down without the use of a reducer since it's two states.
You're either opening it and passing it the modal options, or closing it and nulling those options out no?